sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -2466,34 +2466,66 @@ async Task<Page<DocumentHealthcareResult>> NextPageFunc(string nextLink, int? pa
         #region Analyze Operation
 
         /// <summary>
-        /// Analyze Operation enables the application to have multiple tasks including NER, PII and KPE.
+        /// StartAnalyzeOperationBatchAsync enables the application to have multiple tasks including NER, PII and KPE.
+        /// Accepts a list of strings which are analyzed asynchronously.
         /// For document length limits, maximum batch size, and supported text encoding, see
         /// <a href="https://docs.microsoft.com/azure/cognitive-services/text-analytics/overview#data-limits"/>.
         /// </summary>
-        /// <param name="documents">The documents to analyze.</param>
+        /// <param name="documents">The list of documents to analyze.</param>
         /// <param name="language">The language that the document is written in.</param>
-        /// <param name="options">The additional configurable <see cref="AnalyzeOperationOptions"/> </param>
-        /// <param name="cancellationToken"><see cref="CancellationToken"/></param>
+        /// <param name="options"> The different operations to pass as options.
+        /// You can use it to have multiple tasks to analyze as well as multiple task item per each individual task.
+        /// For example -
+        ///    AnalyzeOperationOptions operationOptions = new AnalyzeOperationOptions()
+        ///    {
+        ///        KeyPhrasesTaskParameters = new KeyPhrasesTaskParameters(),
+        ///        EntitiesTaskParameters = new EntitiesTaskParameters(),
+        ///        PiiTaskParameters = new PiiTaskParameters(),
+        ///        DisplayName = "AnalyzeOperation"
+        ///    };
+        /// By default ModelVersion is set as 'latest' and it can set from the task parameters.
+        /// KeyPhrasesTaskParameters = new KeyPhrasesTaskParameters()
+        /// {
+        ///     ModelVersion = "latest"
+        /// },
+        /// For additional configurable options see <see cref="AnalyzeOperationOptions"/> </param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
         /// <exception cref="RequestFailedException">Service returned a non-success
         /// status code.</exception>
         public virtual async Task<AnalyzeOperation> StartAnalyzeOperationBatchAsync(IEnumerable<string> documents, string language = default, AnalyzeOperationOptions options = default, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNullOrEmpty(documents, nameof(documents));
-            options ??= new AnalyzeOperationOptions();
+            Argument.AssertNotNull(options, nameof(options));
```

Comment:
ahhh I see... so although this is good it won't be enough. The signature of the method still implies that `AnalyzeOperationOptions` is optional, hence `StartAnalyzeOperationBatchAsync(... AnalyzeOperationOptions options = default...)`.

Reply:
@maririos I have made the change for making it mandatory. please include all the commits or the last commit.